packages/components/src/Autocomplete/Autocomplete.types.ts

```diff
@@ -166,3 +229,446 @@ export interface CustomOptionsMenuProp<
+   */
+  readonly renderPersistent?: (args: {
+    value: MenuPersistent<ActionExtra>;
+    position: "header" | "footer";
+    isActive?: boolean;
+  }) => React.ReactNode;
+
+  /*
+   * Render prop to customize the rendering of the input.
+   * @param props.inputRef - The ref to the input element
+   * @param props.inputProps - The props to pass to the input element
+   * Note that you must pass the inputRef to the input
+   */
+  readonly renderInput?: (props: {
+    inputRef: Ref<HTMLInputElement | HTMLTextAreaElement>;
+    inputProps: InputTextRebuiltProps;
+  }) => React.ReactNode;
+
+  readonly UNSAFE_className?: {
+    menu?: string;
+    option?: string;
+    section?: string;
+    action?: string;
+    input?: string;
+    persistentHeader?: string;
+    persistentFooter?: string;
+  };
+
+  readonly UNSAFE_styles?: {
+    menu?: CSSProperties;
+    option?: CSSProperties;
+    section?: CSSProperties;
+    action?: CSSProperties;
+    input?: CSSProperties;
+    persistentHeader?: CSSProperties;
+    persistentFooter?: CSSProperties;
+  };
+
+  /*
+   * Render a custom empty state when the menu is empty.
+   * NOTE: Do not put interactive elements in the empty state, it will break accessibility.
+   * If you require interactive elements in the empty state, use the `emptyActions` prop.
+   * To opt out of the default empty state message entirely use "false".
+   *
+   * @default string "No options"
+   */
+  readonly emptyStateMessage?: React.ReactNode;
+
+  /*
+   * Actions to display when there are no options to render after filtering.
+   * Can be a static list or a function that derives actions from the current input value.
+   * When provided and options are empty, these are rendered as navigable actions. Compatible with or without `emptyStateMessage`.
+   */
+  readonly emptyActions?:
+    | MenuAction<ActionExtra>[]
+    | ((args: { inputValue: string }) => MenuAction<ActionExtra>[]);
+
+  /*
+   * Whether the menu should open when the input gains focus.
+   *
+   * @default true
+   */
+  readonly openOnFocus?: boolean;
+
+  /*
+   * The placeholder text for the input.
+   */
+  readonly placeholder?: string;
+  /*
+   * Whether the input is disabled.
+   */
+  readonly disabled?: boolean;
+  /*
+   * Error message to display below the input
+   * When present, invalid appearance applied to the input
+   */
+  readonly error?: string;
+  /*
+   * Whether the input is invalid. Receives invalid appearance.
+   */
+  readonly invalid?: boolean;
+  /*
+   * Whether the input is read-only.
+   * @default false
+   */
+  readonly readOnly?: boolean;
+  /*
+   * Description to display below the input
+   */
+  readonly description?: string;
+  /*
+   * Name of the input for form submission
+   */
+  readonly name?: string;
+  /*
+   * Size of the input
+   */
+  readonly size?: InputTextRebuiltProps["size"];
+
+  /*
+   * Whether the input is clearable.
+   * TODO: implement (issue with width)
+   */
+  readonly clearable?: boolean;
```

Comment:
I've fixed the layout issue, but something weird happens after a selection. I'll see if I can solve it, but if not this might have to be a follow up